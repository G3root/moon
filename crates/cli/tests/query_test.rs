use insta::assert_snapshot;
use moon_utils::test::{
    create_fixtures_sandbox, create_moon_command_in, get_assert_output, run_git_command,
};

mod touched_files {
    use super::*;

    #[test]
    fn can_change_options() {
        let fixture = create_fixtures_sandbox("cases");

        run_git_command(fixture.path(), "Failed to create branch", |cmd| {
            cmd.args(["checkout", "-b", "branch"]);
        });

        let assert = create_moon_command_in(fixture.path())
            .arg("query")
            .arg("touched-files")
            .args([
                "--base",
                "master",
                "--head",
                "branch",
                "--status",
                "deleted",
                "--upstream",
            ])
            .assert();

        assert_snapshot!(get_assert_output(&assert));
    }
}
