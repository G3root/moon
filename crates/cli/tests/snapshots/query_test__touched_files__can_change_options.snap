---
source: crates/cli/tests/query_test.rs
assertion_line: 31
expression: get_assert_output(&assert)
---
{
  "files": [],
  "options": {
    "base": "master",
    "head": "branch",
    "log": false,
    "status": "deleted",
    "upstream": true
  }
}


